src/VirtualClient/VirtualClient.Monitors/Nvidia-Smi/NvidiaSmiMonitor.cs

```diff
@@ -64,21 +82,53 @@ protected override void Validate()
             }
         }
 
-        private async Task QueryC2CAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        private async Task QueryNvLinkBandwidthAsync(EventContext telemetryContext, CancellationToken cancellationToken)
         {
-            ISystemManagement systemManagement = this.Dependencies.GetService<ISystemManagement>();
+            string arguments = "nvlink -gt d";
+
+            try
+            {
+                DateTime startTime = DateTime.UtcNow;
+                IProcessProxy process = await this.ExecuteCommandAsync(NvidiaSmiCommand, arguments, Environment.CurrentDirectory, telemetryContext, cancellationToken, runElevated: true);
+                DateTime endTime = DateTime.UtcNow;
+
+                if (!cancellationToken.IsCancellationRequested)
+                {
+                    await this.LogProcessDetailsAsync(process, telemetryContext, "Nvidia-Smi-NvLink", logToFile: true);
+                    process.ThrowIfErrored<MonitorException>(errorReason: ErrorReason.MonitorFailed);
+
+                    if (process.StandardOutput.Length > 0)
+                    {
+                        NvidiaSmiQueryNvLinkParser parser = new NvidiaSmiQueryNvLinkParser(process.StandardOutput.ToString());
+                        IList<Metric> metrics = parser.Parse();
+
+                        if (metrics?.Any() == true)
+                        {
+                            this.Logger.LogPerformanceCounters("nvidia", metrics, startTime, endTime, telemetryContext);
+                        }
+                    }
+                }
+            }
+            catch (OperationCanceledException)
```

Comment:
Could we reduce the duplicated codes here all these are calling commands and error handling, only thing that changes are command arguments and parser